hitmap.go

```diff
@@ -125,3 +127,12 @@ func (hits *hitMap) importIPRates(serviceDomains ServiceDomains) error {
 
 	return nil
 }
+
+// Calls our RWMutex Lock(), and prints a warning if we blocked too long.
+func (hits *hitMap) Lock() {
+	ts := time.Now()
+	hits.RWMutex.Lock()
+	if d := ts.Sub(time.Now()); d > time.Duration(1)*time.Second {
```

Comment:
Not interested in a timeout. An after-the-fact log is acceptable for my needs here. Just ensuring lock contention isn't happening anywhere (which has happened in the past).